### PR TITLE
fixes: iPhone padding insets in PWA

### PR DESCRIPTION
**Pull request by raygervais**
## Type of Change

- [X] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

On an iPhone with notch, the bottom toolbar collides with the gesture pill. I really want to fix that so it's a more polished experience in mobile / PWA form.

image

## Checklist

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

**Comment by raygervais** [comment]
Example of fixed bottom inset, now need to fix top since it pushes content up
image

**Comment by raygervais** [comment]
Adds inset support for Navbar on Banner too: 
image

**Comment by raygervais** [comment]
After many trial-error based approaches, I present the fix!
image

Open to your reviews when convenient @PedroFonsecaDEV @humphd

**Review by PedroFonsecaDEV — APPROVED** [review]
LGTM.

@DukeManh

**Comment by raygervais** [comment]
I realize I need to also update the search and about pages. Let me do that before merge

**Comment by humphd** [comment]
I don't have an iPhone to test this, so I'll punt to those who do.  Thanks for doing this!

File: src/web/src/components/NavBar/index.tsx [hunk 1]
```diff
@@ -37,6 +37,7 @@ const useStyles = makeStyles((theme) => ({
       left: '-3vw',
     },
     [theme.breakpoints.down(1024)]: {
+      paddingTop: '0px',
```
Comment (humphd) [inline src/web/src/components/NavBar/index.tsx:40]: 0 vs. 0px please

File: src/web/src/components/BannerButtons.tsx [hunk 2]
```diff
@@ -8,6 +8,7 @@ import PopUp from './PopUp';
 
 const useStyles = makeStyles((theme) => ({
   buttonsContainer: {
+    paddingTop: ' env(safe-area-inset-top, 0px)',
```
Comment (humphd) [inline src/web/src/components/BannerButtons.tsx:11]: 0 vs. 0px please

**Comment by raygervais** [comment]
Still testing & improving implementation, will squash and request review once ready.

**Comment by raygervais** [comment]
Finally ready for review @humphd @PedroFonsecaDEV! 

Very hard to test & debug without an iPhone simulator, so every commit here which triggers the preview I checked on my iPhone in PWA mode. Here are some screenshots: 

image
image
image
image

**Review by PedroFonsecaDEV — APPROVED** [review]
@raygervais 
It seems to be working in all screen sizes on my end. The screenshots were very important too.
Thanks for this Ray.
LGTM.

File: src/web/src/components/BannerButtons.tsx [hunk 3]
```diff
@@ -8,6 +8,7 @@ import PopUp from './PopUp';
 
 const useStyles = makeStyles((theme) => ({
   buttonsContainer: {
+    paddingTop: ' env(safe-area-inset-top, 0)',
```
Comment (birtony) [inline src/web/src/components/BannerButtons.tsx:11]: Looks like there is an extra space prior to `env(...`. Rather than that, looks good on my iPhone!

Comment (raygervais) [reply src/web/src/components/BannerButtons.tsx:11]: Good catch, will fix.

**Comment by raygervais** [comment]
Resolved the extra space @birtony @PedroFonsecaDEV. One last review and then this can be merged

**Review by PedroFonsecaDEV — APPROVED** [review]

**Review by birtony — APPROVED** [review]

**Pull request merged**